core/src/main/scala/org/http4s/Headers.scala

```diff
@@ -87,6 +87,14 @@ final class Headers private (headers: List[Header])
     }
     else super.++(that)
   }
+
+  override def hashCode(): Int = this.headers.hashCode()
+
+  override def equals(that: Any): Boolean = that match {
+    case otherheaders: Headers =>
+      this.headers.equals(otherheaders.toList)
```

Comment:
`headers` and `toList` is [the same](https://github.com/andimiller/http4s/blob/ec7969ad68153bffd83389493c46aa96e9efbc78/core/src/main/scala/org/http4s/Headers.scala#L18), but could you pick one?